src/overlays/ovl_i2/fox_sx.c

```diff
@@ -1,81 +1,1986 @@
+                        gDPSetPrimColor(gMasterDisp++, 0, 0, 255, 255, 255, 255);
+                    } else {
+                        gDPSetPrimColor(gMasterDisp++, 0, 0, 255, 255, 0, 255);
+                    }
+                    Matrix_Push(&gGfxMatrix);
+                    Matrix_Translate(gGfxMatrix, D_i2_80195644.x, D_i2_80195644.y, D_i2_80195644.z, (u8) 1);
+                    Matrix_RotateZ(gGfxMatrix, D_i2_80195650[2] * M_DTOR, 1);
+                    Matrix_RotateY(gGfxMatrix, D_i2_80195650[1] * M_DTOR, 1);
+                    Matrix_RotateX(gGfxMatrix, D_i2_80195650[0] * M_DTOR, 1);
+                    Matrix_RotateZ(gGfxMatrix, 2.0f * gFrameCount * M_DTOR, 1);
+                    Matrix_Scale(gGfxMatrix, boss->fwork[45], boss->fwork[45], boss->fwork[45], 1);
+                    Matrix_SetGfxMtx(&gMasterDisp);
+                    gSPDisplayList(gMasterDisp++, D_6009C30);
+                    Matrix_Pop(&gGfxMatrix);
+                    Matrix_Translate(gGfxMatrix, D_i2_8019565C.x, D_i2_8019565C.y, D_i2_8019565C.z, (u8) 1);
+                    Matrix_RotateZ(gGfxMatrix, D_i2_80195668[2] * M_DTOR, 1);
+                    Matrix_RotateY(gGfxMatrix, D_i2_80195668[1] * M_DTOR, 1);
+                    Matrix_RotateX(gGfxMatrix, D_i2_80195668[0] * M_DTOR, 1);
+                    Matrix_RotateZ(gGfxMatrix, 2.0f * -gFrameCount * M_DTOR, 1);
+                    Matrix_Scale(gGfxMatrix, boss->fwork[45], boss->fwork[45], boss->fwork[45], 1);
+                    Matrix_SetGfxMtx(&gMasterDisp);
+                    gSPDisplayList(gMasterDisp++, D_6009C30);
+                }
+                Matrix_Pop(&gGfxMatrix);
+                Matrix_Pop(&gCalcMatrix);
+            }
+            return 1;
+
+        case 34:
+            arg3->x += boss->fwork[27];
+            break;
+
+        case 33:
+            arg2->z -= boss->fwork[31];
+            arg2->y -= boss->fwork[29];
+            arg2->x -= boss->fwork[30];
+            break;
+
+        case 38:
+            arg3->x += boss->fwork[28];
+            break;
+
+        case 37:
+            arg2->z += boss->fwork[34];
+            arg2->y += boss->fwork[32];
+            arg2->x -= boss->fwork[33];
+            break;
+
+        case 35:
+            arg2->x -= boss->fwork[14];
+            arg2->y -= boss->fwork[16];
+            arg2->z -= boss->fwork[15];
+            if (boss->actionState == 0x14) {
+                arg3->x += boss->fwork[14] * 0.2f;
+                arg3->y += boss->fwork[14] * 0.1f;
+            }
+            break;
+
+        case 39:
+            arg2->x += boss->fwork[17];
+            arg2->y -= boss->fwork[19];
+            arg2->z += boss->fwork[18];
+            if (boss->actionState == 0x14) {
+                arg3->x += boss->fwork[17] * 0.1f;
+                arg3->z += boss->fwork[17] * 0.2f;
+            }
+            break;
+    }
+
+    return 0;
+}
+
+void func_i2_80193208(s32 arg0, Vec3f* arg1, void* ptr) {
```

Comment:
Signature of a post limb draw should be `void func_i2_80193208(s32 limbIndex, Vec3f* rot, void* data)`, where data can be renamed to something more fitting.